client/files.go

```diff
@@ -0,0 +1,226 @@
+			return err
+		}
+
+		if !append || errors.Is(err, services.ErrObjectNotExist) {
+			object, err = appender.CreateAppendWithContext(c.commandAbortCtx, path)
+			if err != nil {
+				return err
+			}
+		}
+
+		_, err = appender.WriteAppendWithContext(c.commandAbortCtx, object, file, size)
+		if err != nil {
+			return err
+		}
+		return nil
+	}
+
+	_, err = c.storager.WriteWithContext(c.commandAbortCtx, path, file, size)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (c *Handler) handleRETR() {
+	defer func() {
+		c.ctxRest = 0
+	}()
+	path := c.absPath(c.param)
+	tr, err := c.TransferOpen()
+	if err != nil {
+		c.WriteMessage(StatusCannotOpenDataConnection, err.Error())
+		return
+	}
+
+	_, err = c.storager.ReadWithContext(c.commandAbortCtx, path, tr, pairs.WithOffset(c.ctxRest))
+	if err != nil {
+		c.TransferClose()
+		c.WriteMessage(StatusActionNotTaken, err.Error())
+		return
+	}
+
+	select {
+	case <-c.commandAbortCtx.Done():
+		c.WriteMessage(StatusTransferAborted, "Connection closed; transfer aborted")
+	default:
+		c.TransferClose()
+		c.WriteMessage(StatusClosingDataConn, "transfer finished")
+	}
+}
+
+func (c *Handler) handleCHMOD(params string) {
+
+}
+
+func (c *Handler) handleDELE() {
+	path := c.absPath(c.param)
+	err := c.storager.DeleteWithContext(c.commandAbortCtx, path)
+	if err != nil {
+		c.WriteMessage(StatusActionNotTaken, fmt.Sprintf("Couldn't delete %s: %v", path, err))
+		return
+	}
+	c.WriteMessage(StatusFileOK, fmt.Sprintf("Removed file %s", path))
+}
+
+func (c *Handler) handleRNFR() {
+	path := c.absPath(c.param)
+	_, err := c.storager.Stat(path)
+	if err != nil {
+		c.WriteMessage(StatusActionNotTaken, fmt.Sprintf("Couldn't access %s: %v", path, err))
+		return
+	}
+	c.WriteMessage(StatusFileActionPending, "Sure, give me a target")
+	c.ctxRnfr = path
+}
+
+func (c *Handler) handleRNTO() {
+	path := c.absPath(c.param)
+	mover, ok := c.storager.(types.Mover)
+	if !ok {
+		c.WriteMessage(StatusCommandNotImplemented, "this type of storage is not support rename")
+		return
+	}
+
+	if c.ctxRnfr == "" {
+		c.WriteMessage(StatusBadCommandSequence, "RNFR is expected before RNTO")
+		return
+	}
+
+	err := mover.Move(c.ctxRnfr, path)
+	if err != nil {
+		c.WriteMessage(StatusActionNotTaken, fmt.Sprintf("Couldn't rename file: %v", err))
+		return
+	}
+
+	c.WriteMessage(StatusFileOK, "Done !")
+	c.ctxRnfr = ""
+}
+
+func (c *Handler) handleSIZE() {
+	path := c.absPath(c.param)
+	object, err := c.storager.Stat(path)
+	if err != nil {
+		c.WriteMessage(StatusActionNotTaken, fmt.Sprintf("Couldn't access %s: %v", path, err))
+		return
+	}
+	length, ok := object.GetContentLength()
+	if !ok {
+		c.WriteMessage(StatusActionNotTaken, fmt.Sprintf("Couldn't access %s: %v", path, err))
+		return
+	}
+	c.WriteMessage(StatusFileStatus, fmt.Sprintf("%d", length))
+}
+
+func (c *Handler) handleSTATFile() {
+	path := c.absPath(c.param)
+
+	c.writeLine("213-Status follows:")
```

Comment:
Maybe it's better to unify all response into `c.WriteMessage`?

Reply:
`c.WriteMessage` accepts a response code and message, it is used for one-line response, warps the  `c.writeLine`. Here, the response is [multiline](http://www.faqs.org/rfcs/rfc959.html#:~:text=Thus%20the%20format,to%20avoid%20confusion.), we need to use `c.writeLine`